### ISSUE TITLE
error plugin fail should return 503 not 500.

### ISSUE THREAD

**Issue by panthershark** [mixdown/plugins#1]

**Comment by cappslock** [comment]
Why? 503 seems better suited to indicate temporary issues such as maintenance or downtime and 500 is more generic, which seems better for the generic "fail" handler.

**Comment by cappslock** [comment]
It seems like it would be better to default to 500 but allow the caller to override the status code,

**Comment by panthershark** [comment]
503 is more generally used to denote an application error.  Since mixdown is targeted as an application framework, then it seems better suited here.  A 500 error is often thrown from a more generic component (such as proxy or load balancer) and changing to 503 allows a user to determine whether a failed response came from their application or not.

**Comment by cappslock** [comment]
I disagree. W3 standards seem pretty clear on the intended use of each code:

```
500 Internal Server Error

The server encountered an unexpected condition which prevented it from fulfilling the request.

10.5.4 503 Service Unavailable

The server is currently unable to handle the request due to a temporary overloading or maintenance of the server. The implication is that this is a temporary condition which will be alleviated after some delay. If known, the length of the delay MAY be indicated in a Retry-After header. If no Retry-After is given, the client SHOULD handle the response as it would for a 500 response.

  Note: The existence of the 503 status code does not imply that a
  server must use it when becoming overloaded. Some servers may wish
  to simply refuse the connection.
```

**Comment by panthershark** [comment]
One problem we have at vast, all the other components in the chain (load balances, pixies, etc) send 500, 502, 504 and none send 503. 503 implies that the service is there (check), the route was handled (check), and that you should try again later which is appropriate (check). 

Your point makes total sense. Restating the problem, there should be a way to tell that a mixdown app was the component that sent the exception. I am all ears.

**Comment by panthershark** [comment]
Pixies.  Proxies. Whatever.

**Comment by cappslock** [comment]
I see. Let me suggest this:
1. Use 500 by default for the fail method of the error plugin.
2. Allow this default to be overridden on plugin init options.
3. Allow the caller of fail to override the default.
4. For tracking the source of the error, add a header to the response indicating it  originated from the mixdown server.

Thoughts on that approach?
